src/lua/mod.rs

```diff
@@ -0,0 +1,283 @@
+        val: Box<hlua::Push<&'static mut Lua<'static>> + Sized>
+    },
+    /// Create a new array
+    EmptyArray(String),
+    /// Message to ping the lua thread
+    Ping,
+    /// Unused send type
+    Unused,
+}
+
+/// Messages received from lua thread
+pub enum LuaResponse {
+    /// Lua variable obtained
+    Variable(Option<AnyLuaValue>),
+    /// Lua error
+    Error(hlua::LuaError),
+    /// A function is returned
+    Function(hlua::functions_read::LuaFunction<String>),
+    /// Pong response from lua ping
+    Pong,
+    /// Unused response type
+    Unused,
+}
+
+unsafe impl Send for LuaQuery { }
+unsafe impl Send for LuaResponse { }
+unsafe impl Sync for LuaQuery { }
+unsafe impl Sync for LuaResponse { }
+
+/// Whether the lua thread is currently available
+pub fn thread_running() -> bool {
+    *RUNNING.read().unwrap()
+}
+
+/// Errors which may arise from attempting
+/// to sending a message to the lua thread.
+pub enum LuaSendError {
+    ThreadClosed,
+    Sender
+}
+
+/// Attemps to send a LuaQuery to the lua thread.
+pub fn try_send(query: LuaQuery) -> Result<(), LuaSendError> {
+    if !thread_running() { Err(LuaSendError::ThreadClosed) }
+    else {
+        match SENDER.lock().unwrap().send(query) {
+            Ok(_) => Ok(()),
+            Err(_) => Err(LuaSendError::Sender)
+        }
+    }
+}
+
+/// Sends a value to the lua thread.
+///
+/// # Panics
+/// * If the lua thread is currently not running (check `lua::thread_running()`)
+/// * If the sender was unable to send, which should only happen
+/// if the lua thread isn't running.
+pub fn send(query: LuaQuery) {
+    if !thread_running() {
+        panic!("lua: Attempted to message the lua thread while it is not active!");
+    }
+    SENDER.lock().unwrap().send(query).unwrap();
+}
+
+/// Initialize the lua thread
+pub fn init() {
+    trace!("Initializing...");
+    let (query_tx, query_rx) = channel::<LuaQuery>();
+    let (answer_tx, answer_rx) = channel::<LuaResponse>();
+    {
+        let mut sender = SENDER.lock().unwrap();
+        let mut receiver = RECEIVER.lock().unwrap();
+
+        *sender = query_tx;
+        *receiver = answer_rx;
+    }
+
+    thread::spawn(move || {
+        trace!("thread: Inside thread!");
+        let sender = answer_tx;
+        let receiver = query_rx;
+        let mut lua = Lua::new();
+        trace!("thread: Loading core libraries...");
+        lua.openlibs();
+        trace!("thread: loading way-cooler library...");
+        lua.execute_from_reader::<(), File>(
+            File::open("lib/lua/init.lua").unwrap()
+        );
+        trace!("thread: loading Rust libraries...");
+        funcs::register_libraries(&mut lua);
+        trace!("Finished loading libraries!");
+        // Only broadcast running after loading libs
+        *RUNNING.write().unwrap() = true;
+        trace!("thread: Testing hello world...");
+        let mut file = File::create("/tmp/init.lua").unwrap();
```

Comment:
Don't assume there is a tmpfs at /tmp/. Perhaps use a crate like _tempfile_ or some other mechanism to have it write to a valid temporary place that the OS can choose.